driver-core/src/main/java/com/datastax/driver/core/Metadata.java

```diff
@@ -134,15 +134,39 @@ static String escapeId(String ident) {
         return lowercaseAlphanumeric.matcher(ident).matches() ? ident : quote(ident);
     }
 
-    // Builds the internal name of a function/aggregate
-    // Note that if simpleName comes from the user, the caller must call handleId on it before passing it to this method
+    /**
+     * Builds the internal name of a function/aggregate, which is similar, but not identical,
+     * to the function/aggregate signature.
+     * This is only used to generate keys for internal metadata maps (KeyspaceMetadata.functions and.
+     * KeyspaceMetadata.aggregates).
+     * Note that if simpleName comes from the user, the caller must call handleId on it before passing it to this method.
+     * Note that this method does not necessarily generates a valid CQL function signature.
+     * Note that argumentTypes can be either a list of strings (schema change events)
+     * or a list of DataTypes (function lookup from client code).
+     * This method must ensure that both cases produce the same identifier.
+     */
     static String fullFunctionName(String simpleName, Collection<?> argumentTypes) {
```

Comment:
The reason why we have this method is that we need an unique identifier that is easily created either with the list of parameter types, or a list of Strings representing the parameter types, as provided by a schema change event. So the algorithm here differs a bit from the one used to create function signatures:
* The function name is not quoted
* user types are not qualified with the keyspace name
* the frozen keyword is allowed